src/hardware/affinity.cpp

```diff
@@ -51,10 +52,14 @@ std::vector<int> get_affinity() {
 }
 #endif
 
-unsigned count_available_cores() {
-    return get_affinity().size();
+util::optional<std::size_t> num_available_cores() {
```

Comment:
I chose `hw` for the namespace because keystrokes.
I chose `hardware` for the path name because it is very clear and unambiguous.

Considerthe name of `num_available_X` shortened.